### ISSUE TITLE
Compilation error / Unknown CMake command "target_link_directories"

### ISSUE THREAD

**Issue by aljazzair** [duncanthrax/scream#118]
Hello,
.
First thank you for creating this software, looking forward to join the community and use it!
.
I am trying to install the Receiver on Centos using the instructions provided, unfortunately it fails.
.
CMake reports Unknown CMake command "target_link_directories".

Any ideas on what is happening?

-------------------------------------------------------------------
Here are the details:

**Environment:**

```
uname -a
Linux netsrvr 4.18.0-193.19.1.el8_2.x86_64 #1 SMP Mon Sep 14 14:37:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
pulseaudio --version
pulseaudio 11.1-rebootstrapped
```

```
cat /proc/asound/version
Advanced Linux Sound Architecture Driver Version k4.18.0-193.19.1.el8_2.x86_64.
```
` cmake --version`

```
cmake version 3.11.4

CMake suite maintained and supported by Kitware (kitware.com/cmake).

```
**Installed dependencies:**

`dnf install pulseaudio-libs-devel alsa-lib-devel `

```firmware for qemu, built by jenkins, fresh from git repos 8.8 kB/s | 2.9 kB     00:00    
Package pulseaudio-libs-devel-11.1-23.el8.x86_64 is already installed.
Package alsa-lib-devel-1.2.1.2-3.el8.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

**Followed the instructions:**

`git clone https://github.com/duncanthrax/scream.git`

```
Cloning into 'scream'...
remote: Enumerating objects: 86, done.
remote: Counting objects: 100% (86/86), done.
remote: Compressing objects: 100% (67/67), done.
remote: Total 748 (delta 40), reused 28 (delta 15), pack-reused 662
Receiving objects: 100% (748/748), 1.19 MiB | 1.16 MiB/s, done.
Resolving deltas: 100% (421/421), done.
```

`cd scream/Receivers/unix/`

 `mkdir build && cd build`
`mkdir: created directory 'build'``

**CMake fails with the following message:**

`cmake ..`

```

-- The C compiler identification is GNU 8.3.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.4.2") 
-- Checking for module 'libpulse-simple'
--   Found libpulse-simple, version 11.1-rebootstrapped
CMake Error at CMakeLists.txt:14 (target_link_directories):
  Unknown CMake command "target_link_directories".

-- Configuring incomplete, errors occurred!
See also "/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeOutput.log".```
```
**Logs:**

`
cat /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeOutput.log`

```
The system is: Linux - 4.18.0-193.19.1.el8_2.x86_64 - x86_64
Compiling the C compiler identification source file "CMakeCCompilerId.c" succeeded.
Compiler: /usr/bin/cc 
Build flags: 
Id flags:  

The output was:
0


Compilation of the C compiler identification source "CMakeCCompilerId.c" produced "a.out"

The C compiler identification is GNU, found in "/home/username/git/scream/Receivers/unix/build/CMakeFiles/3.11.4/CompilerIdC/a.out"

Determining if the C compiler works passed with the following output:
Change Dir: /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_5c5f3/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_5c5f3.dir/build.make CMakeFiles/cmTC_5c5f3.dir/build
gmake[1]: Entering directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_5c5f3.dir/testCCompiler.c.o
/usr/bin/cc    -o CMakeFiles/cmTC_5c5f3.dir/testCCompiler.c.o   -c /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp/testCCompiler.c
Linking C executable cmTC_5c5f3
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_5c5f3.dir/link.txt --verbose=1
/usr/bin/cc      CMakeFiles/cmTC_5c5f3.dir/testCCompiler.c.o  -o cmTC_5c5f3 
gmake[1]: Leaving directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'


Detecting C compiler ABI info compiled with the following output:
Change Dir: /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_30f05/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_30f05.dir/build.make CMakeFiles/cmTC_30f05.dir/build
gmake[1]: Entering directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o
/usr/bin/cc    -o CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o   -c /usr/share/cmake/Modules/CMakeCCompilerABI.c
Linking C executable cmTC_30f05
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_30f05.dir/link.txt --verbose=1
/usr/bin/cc     -v CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o  -o cmTC_30f05 
Using built-in specs.
COLLECT_GCC=/usr/bin/cc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --disable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.3.1 20191121 (Red Hat 8.3.1-5) (GCC) 
COMPILER_PATH=/usr/libexec/gcc/x86_64-redhat-linux/8/:/usr/libexec/gcc/x86_64-redhat-linux/8/:/usr/libexec/gcc/x86_64-redhat-linux/:/usr/lib/gcc/x86_64-redhat-linux/8/:/usr/lib/gcc/x86_64-redhat-linux/
LIBRARY_PATH=/usr/lib/gcc/x86_64-redhat-linux/8/:/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/:/lib/../lib64/:/usr/lib/../lib64/:/usr/lib/gcc/x86_64-redhat-linux/8/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_30f05' '-mtune=generic' '-march=x86-64'
 /usr/libexec/gcc/x86_64-redhat-linux/8/collect2 -plugin /usr/libexec/gcc/x86_64-redhat-linux/8/liblto_plugin.so -plugin-opt=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper -plugin-opt=-fresolution=/tmp/ccbko8kG.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --no-add-needed --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o cmTC_30f05 /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/8/crtbegin.o -L/usr/lib/gcc/x86_64-redhat-linux/8 -L/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/8/../../.. CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/lib/gcc/x86_64-redhat-linux/8/crtend.o /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crtn.o
COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_30f05' '-mtune=generic' '-march=x86-64'
gmake[1]: Leaving directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'


Parsed C implicit link information from above output:
  link line regex: [^( *|.*[/\])(ld|CMAKE_LINK_STARTFILE-NOTFOUND|([^/\]+-)?ld|collect2)[^/\]*( |$)]
  ignore line: [Change Dir: /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp]
  ignore line: []
  ignore line: [Run Build Command:"/usr/bin/gmake" "cmTC_30f05/fast"]
  ignore line: [/usr/bin/gmake -f CMakeFiles/cmTC_30f05.dir/build.make CMakeFiles/cmTC_30f05.dir/build]
  ignore line: [gmake[1]: Entering directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp']
  ignore line: [Building C object CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o]
  ignore line: [/usr/bin/cc    -o CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o   -c /usr/share/cmake/Modules/CMakeCCompilerABI.c]
  ignore line: [Linking C executable cmTC_30f05]
  ignore line: [/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_30f05.dir/link.txt --verbose=1]
  ignore line: [/usr/bin/cc     -v CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o  -o cmTC_30f05 ]
  ignore line: [Using built-in specs.]
  ignore line: [COLLECT_GCC=/usr/bin/cc]
  ignore line: [COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper]
  ignore line: [OFFLOAD_TARGET_NAMES=nvptx-none]
  ignore line: [OFFLOAD_TARGET_DEFAULT=1]
  ignore line: [Target: x86_64-redhat-linux]
  ignore line: [Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --disable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux]
  ignore line: [Thread model: posix]
  ignore line: [gcc version 8.3.1 20191121 (Red Hat 8.3.1-5) (GCC) ]
  ignore line: [COMPILER_PATH=/usr/libexec/gcc/x86_64-redhat-linux/8/:/usr/libexec/gcc/x86_64-redhat-linux/8/:/usr/libexec/gcc/x86_64-redhat-linux/:/usr/lib/gcc/x86_64-redhat-linux/8/:/usr/lib/gcc/x86_64-redhat-linux/]
  ignore line: [LIBRARY_PATH=/usr/lib/gcc/x86_64-redhat-linux/8/:/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/:/lib/../lib64/:/usr/lib/../lib64/:/usr/lib/gcc/x86_64-redhat-linux/8/../../../:/lib/:/usr/lib/]
  ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_30f05' '-mtune=generic' '-march=x86-64']
  link line: [ /usr/libexec/gcc/x86_64-redhat-linux/8/collect2 -plugin /usr/libexec/gcc/x86_64-redhat-linux/8/liblto_plugin.so -plugin-opt=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper -plugin-opt=-fresolution=/tmp/ccbko8kG.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --no-add-needed --eh-frame-hdr --hash-style=gnu -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o cmTC_30f05 /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/8/crtbegin.o -L/usr/lib/gcc/x86_64-redhat-linux/8 -L/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/8/../../.. CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/lib/gcc/x86_64-redhat-linux/8/crtend.o /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crtn.o]
    arg [/usr/libexec/gcc/x86_64-redhat-linux/8/collect2] ==> ignore
    arg [-plugin] ==> ignore
    arg [/usr/libexec/gcc/x86_64-redhat-linux/8/liblto_plugin.so] ==> ignore
    arg [-plugin-opt=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper] ==> ignore
    arg [-plugin-opt=-fresolution=/tmp/ccbko8kG.res] ==> ignore
    arg [-plugin-opt=-pass-through=-lgcc] ==> ignore
    arg [-plugin-opt=-pass-through=-lgcc_s] ==> ignore
    arg [-plugin-opt=-pass-through=-lc] ==> ignore
    arg [-plugin-opt=-pass-through=-lgcc] ==> ignore
    arg [-plugin-opt=-pass-through=-lgcc_s] ==> ignore
    arg [--build-id] ==> ignore
    arg [--no-add-needed] ==> ignore
    arg [--eh-frame-hdr] ==> ignore
    arg [--hash-style=gnu] ==> ignore
    arg [-m] ==> ignore
    arg [elf_x86_64] ==> ignore
    arg [-dynamic-linker] ==> ignore
    arg [/lib64/ld-linux-x86-64.so.2] ==> ignore
    arg [-o] ==> ignore
    arg [cmTC_30f05] ==> ignore
    arg [/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crt1.o] ==> ignore
    arg [/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crti.o] ==> ignore
    arg [/usr/lib/gcc/x86_64-redhat-linux/8/crtbegin.o] ==> ignore
    arg [-L/usr/lib/gcc/x86_64-redhat-linux/8] ==> dir [/usr/lib/gcc/x86_64-redhat-linux/8]
    arg [-L/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64] ==> dir [/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64]
    arg [-L/lib/../lib64] ==> dir [/lib/../lib64]
    arg [-L/usr/lib/../lib64] ==> dir [/usr/lib/../lib64]
    arg [-L/usr/lib/gcc/x86_64-redhat-linux/8/../../..] ==> dir [/usr/lib/gcc/x86_64-redhat-linux/8/../../..]
    arg [CMakeFiles/cmTC_30f05.dir/CMakeCCompilerABI.c.o] ==> ignore
    arg [-lgcc] ==> lib [gcc]
    arg [--as-needed] ==> ignore
    arg [-lgcc_s] ==> lib [gcc_s]
    arg [--no-as-needed] ==> ignore
    arg [-lc] ==> lib [c]
    arg [-lgcc] ==> lib [gcc]
    arg [--as-needed] ==> ignore
    arg [-lgcc_s] ==> lib [gcc_s]
    arg [--no-as-needed] ==> ignore
    arg [/usr/lib/gcc/x86_64-redhat-linux/8/crtend.o] ==> ignore
    arg [/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/crtn.o] ==> ignore
  collapse library dir [/usr/lib/gcc/x86_64-redhat-linux/8] ==> [/usr/lib/gcc/x86_64-redhat-linux/8]
  collapse library dir [/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64] ==> [/usr/lib64]
  collapse library dir [/lib/../lib64] ==> [/lib64]
  collapse library dir [/usr/lib/../lib64] ==> [/usr/lib64]
  collapse library dir [/usr/lib/gcc/x86_64-redhat-linux/8/../../..] ==> [/usr/lib]
  implicit libs: [gcc;gcc_s;c;gcc;gcc_s]
  implicit dirs: [/usr/lib/gcc/x86_64-redhat-linux/8;/usr/lib64;/lib64;/usr/lib]
  implicit fwks: []




Detecting C [-std=c11] compiler features compiled with the following output:
Change Dir: /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_98eef/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_98eef.dir/build.make CMakeFiles/cmTC_98eef.dir/build
gmake[1]: Entering directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_98eef.dir/feature_tests.c.o
/usr/bin/cc   -std=c11 -o CMakeFiles/cmTC_98eef.dir/feature_tests.c.o   -c /home/username/git/scream/Receivers/unix/build/CMakeFiles/feature_tests.c
Linking C executable cmTC_98eef
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_98eef.dir/link.txt --verbose=1
/usr/bin/cc      CMakeFiles/cmTC_98eef.dir/feature_tests.c.o  -o cmTC_98eef 
gmake[1]: Leaving directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'


    Feature record: C_FEATURE:1c_function_prototypes
    Feature record: C_FEATURE:1c_restrict
    Feature record: C_FEATURE:1c_static_assert
    Feature record: C_FEATURE:1c_variadic_macros


Detecting C [-std=c99] compiler features compiled with the following output:
Change Dir: /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_3606d/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_3606d.dir/build.make CMakeFiles/cmTC_3606d.dir/build
gmake[1]: Entering directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_3606d.dir/feature_tests.c.o
/usr/bin/cc   -std=c99 -o CMakeFiles/cmTC_3606d.dir/feature_tests.c.o   -c /home/username/git/scream/Receivers/unix/build/CMakeFiles/feature_tests.c
Linking C executable cmTC_3606d
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_3606d.dir/link.txt --verbose=1
/usr/bin/cc      CMakeFiles/cmTC_3606d.dir/feature_tests.c.o  -o cmTC_3606d 
gmake[1]: Leaving directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'


    Feature record: C_FEATURE:1c_function_prototypes
    Feature record: C_FEATURE:1c_restrict
    Feature record: C_FEATURE:0c_static_assert
    Feature record: C_FEATURE:1c_variadic_macros


Detecting C [-std=c90] compiler features compiled with the following output:
Change Dir: /home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_d1492/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_d1492.dir/build.make CMakeFiles/cmTC_d1492.dir/build
gmake[1]: Entering directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_d1492.dir/feature_tests.c.o
/usr/bin/cc   -std=c90 -o CMakeFiles/cmTC_d1492.dir/feature_tests.c.o   -c /home/username/git/scream/Receivers/unix/build/CMakeFiles/feature_tests.c
Linking C executable cmTC_d1492
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_d1492.dir/link.txt --verbose=1
/usr/bin/cc      CMakeFiles/cmTC_d1492.dir/feature_tests.c.o  -o cmTC_d1492 
gmake[1]: Leaving directory '/home/username/git/scream/Receivers/unix/build/CMakeFiles/CMakeTmp'


    Feature record: C_FEATURE:1c_function_prototypes
    Feature record: C_FEATURE:0c_restrict
    Feature record: C_FEATURE:0c_static_assert
    Feature record: C_FEATURE:0c_variadic_macros
```

**Comment by aljazzair** [comment]
Successfully compiled. Documenting resolution just in case someone has the same issue:

It appears cmake version 3.11.4 included in Centos 8.2  does not support the command "target_link_directories" as it only appears in v3.13+

- https://cmake.org/cmake/help/v3.11/manual/cmake-commands.7.html
- https://cmake.org/cmake/help/v3.12/manual/cmake-commands.7.html
- https://cmake.org/cmake/help/v3.13/manual/cmake-commands.7.html

Therefore I had to build cmake from source (v3.15).

Closing this issue.

**Comment by GlebOlden** [comment]
> Successfully compiled. Documenting resolution just in case someone has the same issue:
> 
> It appears cmake version 3.11.4 included in Centos 8.2 official repositories does not support the command "target_link_directories" as it only appears in v3.13+
> 
> * https://cmake.org/cmake/help/v3.11/manual/cmake-commands.7.html
> * https://cmake.org/cmake/help/v3.12/manual/cmake-commands.7.html
> * https://cmake.org/cmake/help/v3.13/manual/cmake-commands.7.html
> 
> Therefore I had to build cmake from source (v3.15).
> 
> Closing this issue.

Thank you @aljazzair. Your decision helped me a lot
